From b044bd5921e9644c9df9655bef10cee0af730724 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Mon, 3 Apr 2017 12:36:35 +0200
Subject: [PATCH] build: remove package makefile overlay functionality

Recent attempts to use it have shown that it does not work properly
except for a few undocumented cases. It's better to remove this now to
avoid having more people fall into the same trap

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 include/package.mk  | 7 -------
 include/toplevel.mk | 2 +-
 2 files changed, 1 insertion(+), 8 deletions(-)

Index: build_dir/include/package.mk
===================================================================
--- build_dir.orig/include/package.mk
+++ build_dir/include/package.mk
@@ -271,7 +271,14 @@ define Build/DefaultTargets
   endef
 endef
 
+define Build/IncludeOverlay
+  $(eval -include $(wildcard $(TOPDIR)/overlay/*/$(PKG_DIR_NAME).mk))
+  define Build/IncludeOverlay
+  endef
+endef
+
 define BuildPackage
+  $(Build/IncludeOverlay)
   $(eval $(Package/Default))
   $(eval $(Package/$(1)))
 
Index: build_dir/include/scan.mk
===================================================================
--- build_dir.orig/include/scan.mk
+++ build_dir/include/scan.mk
@@ -19,7 +19,7 @@ endef
 ifeq ($(SCAN_NAME),target)
   SCAN_DEPS=image/Makefile profiles/*.mk $(TOPDIR)/include/kernel*.mk $(TOPDIR)/include/target.mk image/*.mk
 else
-  SCAN_DEPS=$(TOPDIR)/include/package*.mk
+  SCAN_DEPS=$(TOPDIR)/include/package*.mk $(TOPDIR)/overlay/*/*.mk
 ifneq ($(call feedname,$(SCAN_DIR)),)
   SCAN_DEPS += $(TOPDIR)/feeds/$(call feedname,$(SCAN_DIR))/*.mk
 endif
